### PR TITLE
upstream LDC changes for test/exceptions/Makefile

### PR DESCRIPTION
**Pull request by thewilsonator**
cc @kinke does this look fine to you?

### CODE REVIEW

File: test/exceptions/Makefile [hunk 1]
```diff
@@ -94,6 +94,10 @@ $(ROOT)/refcounted.done: $(ROOT)/refcounted
 	$(QUIET) $<
 	@touch $@
 
+ifeq (ldmd,$(findstring ldmd,$(HOST_DC)))
+# LDC: Make sure allocation intended to provoke exception is not elided.
+$(ROOT)/invalid_memory_operation: DFLAGS+=-disable-gc2stack
+endif
```
Comment (kinke) [inline test/exceptions/Makefile:100]: I don't think this should be here upstream, way too LDC-specific.

Comment (Geod24) [reply test/exceptions/Makefile:100]: What is the problem with this ?

File: test/exceptions/Makefile [hunk 2]
```diff
@@ -85,7 +85,7 @@ $(ROOT)/rt_trap_exceptions_drt_gdb.done: $(ROOT)/rt_trap_exceptions_drt
 	$(QUIET)$(TIMELIMIT) $(GDB) -n -ex 'set confirm off' -ex run -ex 'bt full' -ex q --args $< --DRT-trapExceptions=0 \
 		> $(ROOT)/rt_trap_exceptions_drt_gdb.output 2>&1 || true
 	cat $(ROOT)/rt_trap_exceptions_drt_gdb.output
-	grep "in D main (args=...) at src/rt_trap_exceptions_drt.d:9" > /dev/null < $(ROOT)/rt_trap_exceptions_drt_gdb.output
+	grep "in D main (args=...) at *.rt_trap_exceptions_drt.d:9" > /dev/null < $(ROOT)/rt_trap_exceptions_drt_gdb.output
```
Comment (kinke) [inline test/exceptions/Makefile:88]: Typo, should be `.*`.

Comment (thewilsonator) [reply test/exceptions/Makefile]: Thanks, fixed.

**Pull request merged**